openwrt/openwrt
# ipq40xx: Add support for Linksys EA6350v3

Specifications:
SOC:    Qualcomm IPQ4018
RAM:    256 MiB Samsung K4B2G1646F-BYK0
FLASH1: MX25L1605D 2 MB
FLASH2:  Winbond W25N01GV 128Mb
ETH:    Qualcomm QCA8075
WLAN1:  Qualcomm Atheros QCA4018 2.4GHz 802.11b/g/n 2x2
WLAN2:  Qualcomm Atheros QCA4018 5GHz 802.11n/ac W2 2x2
INPUT:    WPS, Reset
LED:           Status - Green
SERIAL: Header at J19, Beneath DC Power Jack
                1-VCC ; 2-TX ; 3-RX; 4-GND;
                Serial 115200-8-N-1.
Tested and working:
 - USB
 - LAN Ethernet (Correct MAC-address)
 - WAN Ethernet (Correct MAC-address)
 - 2.4 GHz WiFi (Correct MAC-address)
 - 5 GHz WiFi (Correct MAC-address)
 - Factory installation from Web UI
 - OpenWRT sysupgrade
 - LED
 - WPS Button (No event attached - logread method)
 - Reset Button

Not Working:
 - No known issues

Install via Web UI:

 - Attach to a LAN port on the router.
 - Connect to the Linksys Smart WiFi Page (default 192.168.1.1)
 - Select the connectivity tab on the left
 - In the manual update box on the right;
 - Select browse, and browse to openwrt-ipq40xx-linksys_ea6350v3-squashfs-factory.bin
 - Click update.
 - Read and accept the warning (or not, if you don't want to take the risk)
 - The router LED will start blinking. When the router LED goes solid, you can now navigate to 192.168.1.1 to your new OpenWRT installation.

Recovery (Automatic):
 - If the device fails to boot after install or upgrade, whilst the unit is turned on:
 1 - Wait 1 minute
 2 - Switch Off and Wait 10 seconds
 3 - Switch on
 4 - Repeat steps 1 to 3, 3 times then go to 5.
 5 - U-boot will have now erased the failed update and switched back to the last working firmware - you should be able to access your router on LAN at 192.168.1.1

Recovery (Manual):
- The steps for manual recovery are the same as the generic u-boot tftp client method.


Signed-off-by: Ryan Pannell <ryan@osukl.com>
